rdrf/registry/patients/models.py

```diff
@@ -919,6 +919,11 @@ def save(self, *args, **kwargs):
             self.active = True
 
         self.last_updated_overall_at = timezone.now()
+
+        # Synchronise the patient's working group with the patient's user's working groups
+        if self.user and set(self.working_groups.all()) != set(self.user.working_groups.all()):
```

Comment:
`self.working_groups.all()` is being queried twice here, and might cause unusual behaviour if two `save`s happen at the same time.

Also, might this be better suited to run somewhere other than the `save` method (ie. in `all_forms_valid`, because the Patient object might be saved under different circumstances where triggering queries would cause issues?

Reply:
I'll move it to `all_forms_valid` as it seems to be the place where other sync type logic is being as well